### PR TITLE
fix: fix api inconsistencies

### PR DESCRIPTION
**Pull request by chenshoo**

### CODE REVIEW

File: packages/polaris-core/jest.config.js [hunk 1]
```diff
@@ -5,6 +5,6 @@ module.exports = {
     testEnvironment: 'node',
     projects: ["<rootDir>", "../polaris-test"],
     testMatch: ['**/*.test.ts'],
-    testTimeout: 150000,
+    testTimeout: 1000000,
```
Comment (ronkatz96) [inline packages/polaris-core/jest.config.js]: revert?

File: packages/polaris-core/src/middlewares/snapshot-middleware.ts [hunk 2]
```diff
@@ -26,13 +26,16 @@ export class SnapshotMiddleware {
...
-            return currentPage;
+            try {
+                currentPage =
+                    context.requestHeaders.snapRequest || this.snapshotConfiguration.autoSnapshot
+                        ? await this.calculateCurrentPageInSnapshotProcess(context, result)
+                        : await result.getData(0, await result.totalCount());
+                this.logger.debug('Snapshot middleware finished job', context);
+                return currentPage;
+            } catch (e) {
+                context.errors = context.errors ? [e, ...context.errors] : [e];
```
Comment (itaykl) [inline packages/polaris-core/src/middlewares/snapshot-middleware.ts]: why not to simply push "e" to context.errors?

File: packages/polaris-core/src/plugins/extensions/extensions-listener.ts [hunk 3]
```diff
@@ -27,14 +27,19 @@ export class ExtensionsListener implements GraphQLRequestListener<PolarisGraphQL
             if (!this.shouldAddWarningsToExtensions) {
                 context.returnedExtensions.warnings = undefined;
             }
+            if (context.errors) {
+                const errors: string[] = [];
+                context.errors?.forEach((err) => errors.push(err.toString()));
+                context.returnedExtensions.errors = errors;
```
Comment (itaykl) [inline packages/polaris-core/src/plugins/extensions/extensions-listener.ts]: why did you used this complex cast into array?

File: packages/polaris-nest/jest.config.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ module.exports = {
     preset: 'ts-jest',
     testEnvironment: 'node',
     testMatch: ['**/*.test.ts'],
-    testTimeout: 15000000,
+    testTimeout: 1000000,
```
Comment (itaykl) [inline packages/polaris-nest/jest.config.js]: revert?

File: packages/polaris-test/jest.config.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ module.exports = {
     preset: 'ts-jest',
     testEnvironment: 'node',
     testMatch: ['**/*.test.ts'],
-    testTimeout: 150000,
+    testTimeout: 1000000,
```
Comment (itaykl) [inline packages/polaris-test/jest.config.js]: revert?

File: packages/polaris-test/shared-resources/logger.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { LoggerConfiguration, LoggerLevel, PolarisGraphQLLogger } from '@enigmatis/polaris-core';
 
 const loggerConfig: LoggerConfiguration = {
-    loggerLevel: LoggerLevel.WARN,
+    loggerLevel: LoggerLevel.INFO,
```
Comment (itaykl) [inline packages/polaris-test/shared-resources/logger.ts]: revert?

File: packages/polaris-typeorm/jest.config.js [hunk 7]
```diff
@@ -3,7 +3,7 @@ module.exports = {
     testEnvironment: 'node',
     clearMocks: true,
     moduleFileExtensions: ['ts', 'js'],
-    testTimeout: 15000000,
+    testTimeout: 1000000,
```
Comment (itaykl) [inline packages/polaris-typeorm/jest.config.js]: revert?

File: packages/polaris-core/src/middlewares/snapshot-middleware.ts (hunk 2 above)
Comment (itaykl) [reply packages/polaris-core/src/middlewares/snapshot-middleware.ts]: if context errors is undefined? I don't want to initialize it with an empty array, just so I could push to it

File: packages/polaris-core/src/plugins/extensions/extensions-listener.ts (hunk 3 above)
Comment (itaykl) [reply packages/polaris-core/src/plugins/extensions/extensions-listener.ts]: cause errors on the context object are of type polarisError, and I want to convert it to string array, so I could push it to the metadata buffer

**Comment by chenshoo** [comment]
refactored snapResponse not to return pagesIds.
refactored globalDataVersion extension to be dataVersion
fixed empty data in pages to return as empty array instead of empty object
fixed return message on a page that doesn't exist to be 'Snapshot page with id ${id} not found' instead of empty object
fixed return message on a snapshot metadata that doesn't exist to be 'Snapshot metadata with id ${id} not found' instead of empty object
refactored errors, irrelevant entities and warnings to be of buffer type
fixed getData in pagination so it would fetch by startIndex and pageSize as it is supposed to do, and not by startIndex and endIndex

**Review by ronkatz96 — APPROVED** [review]

**Pull request merged**